openedx/core/djangoapps/catalog/docs/decisions/001-programs-cache.rst

```diff
@@ -3,7 +3,15 @@ _____________________
 
 Status
 ======
-Pending
+Accepted
+
+Decision
+=========
+1. Developers working on code involving programs (which is not limited Programs theme developers) should be directed to this ADR for context on the usage and limitations of the programs cache.
+2. Consumers of the programs cache should expect cache misses and either handle them gracefully when possible or explicitly fail (with logging).
+    a. Depending on the cache access pattern, it may be impossible to distinguish between a non-existent program and an existent program that is missing from the cache.
+3. Introduce logging to the programs cache, especially in the case of cache misses or an empty cache. This will make debugging easier and make it more apparent when intervention (such as manually running cache_programs) is necessary.
```

Comment:
Should this also include adding alerting to the cache misses to notify the relevant team?

Reply:
What would be the most relevant team? I can think of making a case for at least 3 separate teams.